### DISCUSSION TITLE
option "-y"

### DISCUSSION THREAD

**Discussion by habub** [merbanan/rtl_433#2204] [Q&A] [answered]
I'm not sure how the -y operator works or what it's supposed to do.
I assume you need input data (sdr or file)  and a decoder (-X or -R) in order to use -y. In my case, regardless of the input, or lack of it, I always get the same output, which is whatever I defined by -y, 
Here is an example of -y without any input (file or radio). What am I missing?

>>>rtl_433 -X "n=Test,m=OOK_PWM,s=128,l=256,r=2576,g=0,t=30,y=380" -y "{23} 00 f2"
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2022-10-17 21:23:41
model     : Test         count     : 1             num_rows  : 1             rows      :
len       : 23           data      : 00f200
codes     : {23}00f200

The full command would be:
>>>rtl_433_64bit_static.exe -r g007_125.15M_250k.cu8 -R 0 -A -X "n=Test,m=OOK_PWM,s=128,l=256,r=2576,g=0,t=30,y=380" -y "{23} 00 f2"

Also, is it possible to use -y on partial data? Not the entire decoded data, but part of it?
Thanks

**Comment by zuckschwerdt** [comment, marked as answer]
Note that `-X` (output mostly raw bitbuffers) is roughly the opposite of `-y` (input raw bitbuffers).
It's used to verify decoding of test data, i.e. input a bitbuffer and get a JSON or KV output.

Any command should only ever use one input source, e.g. `-r` or `-y`, not both.
Also give arguments in in order of "processing flow", e.g.:
`rtl_433 -R 0 -X "n=Test,m=OOK_PWM,s=128,l=256,r=2576,g=0,t=30,y=380" -A -r g007_125.15M_250k.cu8`
(disable decoders, add a decoder, activate analyzer, read input)

**Comment by habub** [comment]
I though -y is an output operator. I should use the "match" option in the flex decoder (-X). Thanks.
